CommonTools/PileupAlgos/src/PuppiAlgo.cc

```diff
@@ -226,3 +227,31 @@ double PuppiAlgo::compute(std::vector<double> const &iVals, double iChi2) const
   lPVal *= ROOT::Math::chisquared_cdf(lVal, lNDOF);
   return lPVal;
 }
+// ------------------------------------------------------------------------------------------
+void PuppiAlgo::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+  fillDescriptionsPuppiAlgo(desc);
+  descriptions.addDefault(desc);
+}
+
+void PuppiAlgo::fillDescriptionsPuppiAlgo(edm::ParameterSetDescription &desc) {
+  desc.add<std::vector<double>>("etaMin", {0.});
+  desc.add<std::vector<double>>("etaMax", {2.5});
+  desc.add<std::vector<double>>("ptMin", {0.});
+  desc.add<std::vector<double>>("MinNeutralPt", {0.2});
+  desc.add<std::vector<double>>("MinNeutralPtSlope", {0.015});
+  desc.add<std::vector<double>>("RMSEtaSF", {1.0});
+  desc.add<std::vector<double>>("MedEtaSF", {1.0});
+  desc.add<double>("EtaMaxExtrap", 2.0);
+
+  edm::ParameterSetDescription puppialgos;
```

Comment:
In the automatically produced puppiProducer_cfi.py you can verify that what follows would include the needed parameterSet in the config. Otherwise you don't have it
```C++
  edm::ParameterSetDescription puppialgos;
  puppialgos.add<int>("algoId", 5);
  puppialgos.add<bool>("useCharged", false);
  puppialgos.add<bool>("applyLowPUCorr", false);
  puppialgos.add<int>("combOpt", 5);
  puppialgos.add<double>("cone", .4);
  puppialgos.add<double>("rmsPtMin", .1);
  puppialgos.add<double>("rmsScaleFactor", 1.0);
  std::vector<edm::ParameterSet> VPSetPuppiAlgos;
  edm::ParameterSet puppiset;
  puppiset.addParameter<int>("algoId", 5);
  puppiset.addParameter<bool>("useCharged", false);
  puppiset.addParameter<bool>("applyLowPUCorr", false);
  puppiset.addParameter<int>("combOpt", 5);
  puppiset.addParameter<double>("cone", .4);
  puppiset.addParameter<double>("rmsPtMin", .1);
  puppiset.addParameter<double>("rmsScaleFactor", 1.0);
  VPSetPuppiAlgos.push_back(puppiset);
  desc.addVPSet("puppiAlgos", puppialgos, VPSetPuppiAlgos);
```